### PR TITLE
bpo-20181: Convert the readline module to the Argument Clinic

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue20181

### CODE REVIEW

File: Modules/readline.c [hunk 1]
```diff
@@ -287,57 +321,51 @@ append_history_file(PyObject *self, PyObject *args)
...
 
-static PyObject*
-set_history_length(PyObject *self, PyObject *args)
+/*[clinic input]
+readline.set_history_length
+
+    length: int
+    /
+
+set the maximal number of lines which will be written to the history file.
```
Comment (serhiy-storchaka) [inline Modules/readline.c]:
```suggestion
Set the maximal number of lines which will be written to the history file.
```

File: Modules/readline.c [hunk 2]
```diff
@@ -287,57 +321,51 @@ append_history_file(PyObject *self, PyObject *args)
...
+
+A negative length is used to inhibit
+history truncation.
```
Comment (serhiy-storchaka) [inline Modules/readline.c]:
```suggestion
A negative length is used to inhibit history truncation.
```

File: Modules/readline.c [hunk 3]
```diff
@@ -287,57 +321,51 @@ append_history_file(PyObject *self, PyObject *args)
...
-set the maximal number of lines which will be written to\n\
-the history file. A negative length is used to inhibit\n\
-history truncation.");
+/* Get history length */
 
+/*[clinic input]
+readline.get_history_length
 
-/* Get history length */
+return the maximum number of lines that will be written to the history file.
```
Comment (serhiy-storchaka) [inline Modules/readline.c]:
```suggestion
Return the maximum number of lines that will be written to the history file.
```

File: Modules/readline.c [hunk 4]
```diff
@@ -375,90 +418,114 @@ set_completion_display_matches_hook(PyObject *self, PyObject *args)
...
-PyDoc_STRVAR(doc_get_completion_type,
-"get_completion_type() -> int\n\
-Get the type of completion being attempted.");
+/* Get the beginning index for the scope of the tab-completion */
 
+/*[clinic input]
+readline.get_begidx
 
-/* Get the beginning index for the scope of the tab-completion */
+get the beginning index of the completion scope
```
Comment (serhiy-storchaka) [inline Modules/readline.c]:
```suggestion
Get the beginning index of the completion scope.
```

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

**Comment by ZackerySpytz** [comment]
I have made the requested changes; please review again.

File: Modules/readline.c [hunk 5]
```diff
@@ -379,90 +421,114 @@ set_completion_display_matches_hook(PyObject *self, PyObject *args)
...
-PyDoc_STRVAR(doc_get_begidx,
-"get_begidx() -> int\n\
-get the beginning index of the completion scope");
+/* Get the ending index for the scope of the tab-completion */
 
+/*[clinic input]
+readline.get_endidx
 
-/* Get the ending index for the scope of the tab-completion */
+get the ending index of the completion scope
```
Comment (serhiy-storchaka) [inline Modules/readline.c]: Please fix also other docstrings.
```suggestion
Get the ending index of the completion scope.
```

File: Modules/readline.c [hunk 6]
```diff
@@ -601,60 +678,75 @@ py_add_history(PyObject *self, PyObject *string)
...
-PyDoc_STRVAR(doc_set_completer,
-"set_completer([function]) -> None\n\
-Set or remove the completer function.\n\
-The function is called as function(text, state),\n\
-for state in 0, 1, 2, ..., until it returns a non-string.\n\
-It should return the next possible completion starting with 'text'.");
+/*[clinic input]
+readline.get_completer
 
+Returns current completer function.
```
Comment (serhiy-storchaka) [inline Modules/readline.c]: Maybe use "get" for uniformity with other functions?

At least "return" is more preferable than "returns".

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
:+1:

**Pull request merged**